web/api/Makefile

```diff
@@ -1,14 +1,63 @@
-# Check if certain environmental variables are set properly.
-include Makefile.check
+THRIFT_OPTS = -r -o /data
+TARGET_PY = --gen py
+TARGET_JS = --gen js:jquery --gen js:node
 
-# Execute the build of each API subfolder.
-SUBDIRS := $(wildcard */.)
+current_dir = $(shell pwd)
+uid = $(shell id -u)
+guid = $(shell id -g)
 
-default: all
+API_DIR := $(current_dir)
 
-all: $(SUBDIRS)
+PY_API_DIR = "$(API_DIR)/py/codechecker_api/codechecker_api"
+PY_API_SHARED_DIR = "$(API_DIR)/py/codechecker_api_shared/codechecker_api_shared"
+JS_API_DIR = "$(API_DIR)/js/codechecker-api-js/lib"
+NODE_JS_API_DIR = "$(API_DIR)/js/codechecker-api-node/lib"
 
-$(SUBDIRS):
-	VERSION=$(dir $@) $(MAKE) -C $@
+default: build
 
-.PHONY: all $(SUBDIRS)
+install_shared:
+	cd codechecker_api_shared && python setup.py install
+
+install_api:
+	cd codechecker_api && python setup.py install
+
+install: install_shared install_api
+
+target_dirs:
+	mkdir -p $(PY_API_DIR)
+	mkdir -p $(PY_API_SHARED_DIR)
+	mkdir -p $(JS_API_DIR)
+	mkdir -p $(NODE_JS_API_DIR)
+
+build: clean target_dirs
+
+	docker run -u "$(uid):$(guid)" -v $(API_DIR):/data thrift:0.9.2 thrift $(THRIFT_OPTS) $(TARGET_PY) $(TARGET_JS) /data/authentication.thrift
```

Comment:
Do we want to start a new container every time we would like to build a thrift file? Couldn't we build the thrift files in one container. For example you can use the following command:
```make
	docker run \
	  -u "$(uid):$(guid)" \
	  -v $(API_DIR):/data \
	  thrift:0.9.2 \
	  /bin/bash -c " \
		thrift $(THRIFT_OPTS) $(TARGET_PY) $(TARGET_JS) /data/authentication.thrift && \
		thrift $(THRIFT_OPTS) $(TARGET_PY) $(TARGET_JS) /data/products.thrift && \
		thrift $(THRIFT_OPTS) $(TARGET_PY) $(TARGET_JS) /data/report_server.thrift && \
		thrift $(THRIFT_OPTS) $(TARGET_PY) $(TARGET_JS) /data/configuration.thrift"
```